lib/usdUfe/ufe/Utils.cpp

```diff
@@ -1349,4 +1389,20 @@ UsdSceneItem::Ptr getParentMaterial(const UsdSceneItem::Ptr& item)
     return prim.GetTypeName() == kMaterial ? UsdSceneItem::create(path, prim) : nullptr;
 }
 
+void setExtractTRSFn(ExtractTRSFn fn)
+{
+    // This function is allowed to be null in which case, a default
+    // implementation will be used.
+    gExtractTRSFn = fn;
+}
+
+void extractTRS(const Ufe::Matrix4d& m, Ufe::Vector3d* t, Ufe::Vector3d* r, Ufe::Vector3d* s)
+{
+    if (gExtractTRSFn) {
+        gExtractTRSFn(m, t, r, s);
+    } else {
+        UsdUfe::internal::getTRS(m, t, r, s);
+    }
```

Comment:
We override this from MayaUsd with a Maya version to extract the TRS. There is a default implementation provided for USD.